src/main/java/net/minecraftforge/client/animation/EntityAnimation.java

```diff
@@ -0,0 +1,174 @@
+    private final Priority priority;
+
+    public EntityAnimation(Mode mode, Priority priority)
+    {
+        this.mode = mode;
+        this.priority = priority;
+    }
+
+    public EntityAnimation(Mode mode)
+    {
+        this(mode, Priority.DEFAULT);
+    }
+
+    /**
+     * Gets the mode of the animation
+     */
+    public final Mode getMode()
+    {
+        return mode;
+    }
+
+    /**
+     * Gets the priority of the animation
+     */
+    public final Priority getPriority()
+    {
+        return priority;
+    }
+
+    /**
+     * The condition to test to determine if the animation can be executed. This is where a test like
+     * holding or using a particular item, riding an entity, etc should be used. There should always
+     * be a condition and never just true.
+     *
+     * @param entity the entity the animation is being applied to
+     * @return the result of the test
+     */
+    public abstract boolean canRun(E entity);
+
+    /**
+     * Executes and applies the animation to the specified model. The model will need to be cast to
+     * the entity specific model in order to access additional fields and methods.
+     *
+     * <p>
+     * For players this would look like:
+     * <p>
+     * <code>if (model instanceof PlayerModel<?> playerModel) {</code>
+     * <p>
+     * <code>// your animation code</code>
+     * <p>
+     * <code>}</code>
+     * <p>
+     * The context parameter contains additional data for calculating animations should it be used.
+     * It should be noted that another animation may have already been applied before this animation
+     * is executed. See {@link EntityAnimation#compareTo(EntityAnimation)} for an explanation on
+     * order animations are applied.
+     *
+     * @param entity the entity currently being rendered
+     * @param root the root model component contain
+     * @param context additional data used for calculating animations
+     */
+    public abstract void apply(E entity, ModelComponent root, Context context);
+
+    /**
+     * Determines how animations are executed. The order is based on the mode and priority
+     * of the animation. Below is a visual representation of the ordering.
+     * <ul>
+     * <li>PASSIVE - FIRST</li>
+     * <li>PASSIVE - DEFAULT</li>
+     * <li>PASSIVE - LAST</li>
+     * <li>ACTIVE - FIRST</li>
+     * <li>ACTIVE - DEFAULT (Cancelled if an ACTIVE - FIRST is executed)</li>
+     * <li>ACTIVE - LAST (Cancelled if an ACTIVE - FIRST/DEFAULT is executed)</li>
+     * </ul>
+     * @param o an entity animation
+     * @return See {@link Comparable} documentation
+     */
+    @Override
+    public final int compareTo(@NotNull EntityAnimation<E> o)
+    {
+        if (this.getMode() == o.getMode())
+        {
+            return this.getPriority().ordinal() - o.getPriority().ordinal();
+        }
+        else if (o.getMode() == Mode.ACTIVE)
+        {
+            return -1;
+        }
+        return 1;
```

Comment:
This can be one line.
```suggestion
        return this.getMode() == o.getMode() ? this.getPriority().ordinal() - o.getPriority().ordinal() : o.getMode() == Mode.ACTIVE ? -1 : 1;
```

Reply:
Why does it need to be? Nested ternaries lead to code that is confusing to read and doesn't seem necessary here.

Reply:
Don't make unreadable code for the sake of unreadability. This can be multiple lines just fine.